tests/foreman/ui_airgun/test_subscription.py

```diff
@@ -0,0 +1,309 @@
+"""Test module for Subscriptions/Manifests UI
+
+:Requirement: Subscription
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: UI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from tempfile import mkstemp
+
+from airgun.session import Session
+from fauxfactory import gen_string
+from nailgun import entities
+from pytest import skip
+
+from robottelo import manifests
+from robottelo.api.utils import create_role_permissions
+from robottelo.cli.factory import make_virt_who_config, virt_who_hypervisor_config
+from robottelo.config import settings
+from robottelo.constants import (
+    DEFAULT_SUBSCRIPTION_NAME,
+    DISTRO_RHEL7,
+    VDC_SUBSCRIPTION_NAME,
+    VIRT_WHO_HYPERVISOR_TYPES,
+)
+from robottelo.decorators import (
+    run_in_one_thread,
+    skip_if_bug_open,
+    skip_if_not_set,
+    setting_is_set,
+    tier2,
+    tier3,
+    upgrade,
+)
+from robottelo.products import RepositoryCollection, RHELAnsibleEngineRepository
+from robottelo.vm import VirtualMachine
+
+pytestmark = [run_in_one_thread]
+
+if not setting_is_set('fake_manifest'):
+    skip('skipping tests due to missing fake_manifest settings', allow_module_level=True)
+
+
+@tier2
+@upgrade
+def test_positive_end_to_end(session, module_org):
+    """Upload a manifest with minimal input parameters, attempt to
+    delete it with checking the warning message and hit 'Cancel' button after than delete it.
+
+    :id: 580fc072-01e0-4f83-8cbb-2a8522d76243
+
+    :expectedresults:
+        1. The manifest was uploaded successfully.
+        2. When attempting to delete the manifest the confirmation dialog contains informative
+            message which warns user about downsides and consequences of manifest deletion.
+        3. When hitting cancel the manifest was not deleted.
+        4. When deleting and confirming deletion, the manifest was deleted successfully.
+
+    :BZ: 1266827
+
+    :CaseImportance: Critical
+    """
+    expected_message_lines = [
+        'Are you sure you want to delete the manifest?',
+        'Note: Deleting a subscription manifest is STRONGLY discouraged. '
+        'Deleting a manifest will:',
+        'Delete all subscriptions that are attached to running hosts.',
+        'Delete all subscriptions attached to activation keys.',
+        'Disable Red Hat Insights.',
+        'Require you to upload the subscription-manifest and re-attach '
+        'subscriptions to hosts and activation keys.',
+        'This action should only be taken in extreme circumstances or for '
+        'debugging purposes.',
+    ]
+    org = entities.Organization().create()
+    _, temporary_local_manifest_path = mkstemp(prefix='manifest-', suffix='.zip')
```

Comment:
doesn't https://github.com/SatelliteQE/robottelo/blob/master/robottelo/manifests.py#L37 do same job?

Reply:
no that's a linux path used for remote manifest upload, but here we need to save it locally and local can windows , linux or mac ...

Reply:
Got your question :
it does not save it to a file the function return a file like content , we need to save it, to use it after for the UI 
```
:return: A file-like object (``BytesIO`` on Python 3 and
            ``StringIO`` on Python 2) with the contents of the cloned
            manifest.
```